src/controllers/sound/MiniaudioBackend.cpp

```diff
@@ -0,0 +1,300 @@
+            BenchmarkGuard b("init sounds");
+
+            ma_uint32 soundFlags = 0;
+            // Decode the sound during loading instead of during playback
+            soundFlags |= MA_SOUND_FLAG_DECODE;
+            // Disable pitch control (we don't use it, so this saves some performance)
+            soundFlags |= MA_SOUND_FLAG_NO_PITCH;
+            // Disable spatialization control, this brings the volume up to "normal levels"
+            soundFlags |= MA_SOUND_FLAG_NO_SPATIALIZATION;
+
+            auto decoderConfig =
+                ma_decoder_config_init(ma_format_f32, 0, 48000);
+            // This must match the encoding format of our default ping sound
+            decoderConfig.encodingFormat = ma_encoding_format_wav;
+
+            for (auto i = 0; i < NUM_SOUNDS; ++i)
+            {
+                auto dec = std::make_unique<ma_decoder>();
+                auto snd = std::make_unique<ma_sound>();
+
+                result = ma_decoder_init_memory(
+                    (void *)this->defaultPingData.data(),
+                    this->defaultPingData.size() * sizeof(char), &decoderConfig,
+                    dec.get());
+                if (result != MA_SUCCESS)
+                {
+                    qCWarning(chatterinoSound) << "Error initializing default "
+                                                  "ping decoder from memory:"
+                                               << result;
+                    return;
+                }
+
+                result = ma_sound_init_from_data_source(this->engine.get(),
+                                                        dec.get(), soundFlags,
+                                                        nullptr, snd.get());
+                if (result != MA_SUCCESS)
+                {
+                    qCWarning(chatterinoSound)
+                        << "Error initializing default sound from data source:"
+                        << result;
+                    return;
+                }
+
+                this->defaultPingDecoders.emplace_back(std::move(dec));
+                this->defaultPingSounds.emplace_back(std::move(snd));
+            }
+        }
+
+        qCInfo(chatterinoSound) << "miniaudio sound system initialized";
+
+        this->initialized = true;
+    });
+
+    this->audioThread = std::make_unique<std::thread>([this] {
+        this->ioContext.run();
+    });
+}
+
+MiniaudioBackend::MiniaudioBackend()
+    : context(std::make_unique<ma_context>())
+    , engine(std::make_unique<ma_engine>())
+    , workGuard(boost::asio::make_work_guard(this->ioContext))
+    , sleepTimer(this->ioContext)
+{
+    qCInfo(chatterinoSound) << "Initializing miniaudio sound backend";
+}
+
+MiniaudioBackend::~MiniaudioBackend()
+{
+    // NOTE: This destructor is never called because the `runGui` function calls _exit before that happens
+    // I have manually called the destructor prior to _exit being called to ensure this logic is sound
```

Comment:
Is it intetional, that it ends in `sound` here?

Reply:
Yes :sunglasses: